### PR TITLE
BUG: Try to coerce index into datetime before failing

### LINKED ISSUE #1180: Try to coerce index into datetime before failing.
See what we can do for these types of problems (maybe with a warning about the copy/conversion).

http://stackoverflow.com/questions/19892830/issue-with-dates-in-statsmodels-and-pandas

### PR DESCRIPTION
**Pull request by bfcondon**
Fixes #1180

TimeSeriesModel tries to coerce pandas index to datetime before throwing an error.

### CODE REVIEW

**Comment by bfcondon** [comment]
I believe all of the tests pass now, but one of the builds crashed.

File: statsmodels/tsa/base/tsa_model.py [hunk 1]
```diff
@@ -48,8 +48,11 @@ def _init_dates(self, dates, freq):
         if dates is not None:
             if (not datetools._is_datetime_index(dates) and
                     isinstance(self.data, data.PandasData)):
-                raise ValueError("Given a pandas object and the index does "
-                                 "not contain dates")
+                try:
+                    dates = to_datetime(dates)
+                except:
```
Comment (josef-pkt) [inline statsmodels/tsa/base/tsa_model.py]: do you have a specific exception that can be put in here?
Blanket exceptions can hide too many problems

**Comment by josef-pkt** [comment]
I had a quick look.
Looks good AFAICS
specific exception?

One of the travisci versions has been randomly getting stuck in a test for around a week now. Unrelated to this.

**Comment by bfcondon** [comment]
@josef-pkt I added a ValueError to the except block.  All tests are now passing.

**Comment by josef-pkt** [comment]
@bfcondon  Thanks looks good to me

can you squash it and force push?

**Comment by josef-pkt** [comment]
one more hint for next time: adding "closes #xxx" where xxx is the issue number will automatically close the relevant issue when merging.

successfully squashed, all green, merging

@bfcondon  Thanks again, It's good to see PRs that include unit tests right away.

**Pull request merged**